example_configs/ocis

```diff
@@ -0,0 +1,99 @@
+# OCIS (OwnCloud Infinite Scale)
+
+This is using version 5 which is currently still in RC.
+
+IMPORTANT: There is a bug/quirk in how the OCIS container handles bind mounts.  
+
+If the bind mount locations (eg. `/srv/ocis/{app,cfg}`) don't exist when the container is started, OCIS creates them with `root` permissions.  It then seems to drop permissions to UID 1000 and gives an error because it can't create files in the `{app,cfg}`.
+
+So you must create the bind mount locations and manually chown them to uid/gid 1000, eg.
+
+```
+# cd /srv/ocis
+# mkdir app cfg
+# chown 1000:1000 app cfg
+# docker compose up -d && docker compose logs -f 
+```
+
+## .env
+
+```
+OCIS_URL="https://ocis.example.nz"
+LDAP_BASE_DN="dc=example,dc=nz"
+LDAP_BIND_PASSWORD=very-secret-yogurt
+# LLDAP UUID to be given admin permissions
+LLDAP_ADMIN_UUID=c1c2428a-xxxx-yyyy-zzzz-6cc946bf6809
+```
+
+## docker-compose.yml
+
+```
+version: "3.7"
+
+networks:
+  caddy:
+    external: true
+
+services:
+  ocis:
+    image: owncloud/ocis:5.0.0-rc.4
+    container_name: ocis
+    networks:
+      - caddy
+    entrypoint:
+      - /bin/sh
+    command: ["-c", "ocis init || true; ocis server"]
+    environment:
+      OCIS_URL: ${OCIS_URL}
+      OCIS_LOG_LEVEL: warn
+      OCIS_LOG_COLOR: "false"
+      PROXY_TLS: "false" # do not use SSL between Traefik and oCIS
+      OCIS_INSECURE: "false"
+      # Basic Auth is required for WebDAV clients that don't support OIDC
+      PROXY_ENABLE_BASIC_AUTH: "false"
+      #IDM_ADMIN_PASSWORD: "${ADMIN_PASSWORD}" # Not needed if admin user is in LDAP (?)
+      #OCIS_PASSWORD_POLICY_BANNED_PASSWORDS_LIST: "banned-password-list.txt"
+
+      # Assumes your LLDAP container is named `lldap`
+      OCIS_LDAP_URI: ldap://lldap:3890
+      OCIS_LDAP_INSECURE: "true"
+      OCIS_LDAP_BIND_DN: "uid=admin,ou=people,${LDAP_BASE_DN}"
+      OCIS_LDAP_BIND_PASSWORD: ${LDAP_BIND_PASSWORD}
+      OCIS_ADMIN_USER_ID: ${LLDAP_ADMIN_UUID}
+
+      OCIS_LDAP_USER_ENABLED_ATTRIBUTE: uid
+      GRAPH_LDAP_SERVER_WRITE_ENABLED: "false" # Does your LLDAP bind user have write access?
+      GRAPH_LDAP_REFINT_ENABLED: "false"
+      # Disable the built in LDAP server
+      OCIS_EXCLUDE_RUN_SERVICES: idm
+      # both text and binary cause errors in LLDAP, seems harmless though (?)
```

Comment:
What's the error you see? I'm curious

Reply:
```
lldap  | 2024-02-19T05:47:32.478666428Z 2024-02-19T05:47:32.478151155+00:00  WARN        ┕━ 🚧 [warn]: Ignoring unrecognized group attribute: text\n\
lldap  | 2024-02-19T05:47:32.478673132Z                       To disable this warning, add it to "ignored_user_attributes" in the config.
```

As far as I can tell from the docs, there's only two possible options `text` and `binary`.  Both generate the same error, but `binary` doesn't work for auth.

Reply:
Can you enable verbose mode? I'd like to see the LDAP query that results into this. It seems like it's just listing "text" as an extra attribute, I wonder how they expect it to work.

Reply:
```
lldap  | 2024-02-19T18:56:32.279688737Z 2024-02-19T18:56:32.189192027+00:00  INFO     ┝━ LDAP request [ 1.43ms | 0.10% / 0.67% ]
lldap  | 2024-02-19T18:56:32.279694940Z 2024-02-19T18:56:32.189211273+00:00  DEBUG    │  ┝━ 🐛 [debug]:  | msg: LdapMsg { msgid: 2, op: SearchRequest(LdapSearchRequest { base: "ou=people,dc=example,dc=nz", scope: Subtree, aliases: Never, sizelimit: 1, timelimit: 60, typesonly: false, filter: And([And([Equality("objectclass", "inetOrgPerson"), And([Equality("objectclass", "person"), Equality("memberOf", "cn=ocis_users,ou=groups,dc=example,dc=nz")]), Not(Equality("uid", "FALSE"))]), Equality("uid", "adam")]), attrs: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"] }), ctrl: [] }
lldap  | 2024-02-19T18:56:32.279729418Z 2024-02-19T18:56:32.189213744+00:00  DEBUG    │  ┝━ do_search [ 1.22ms | 0.13% / 0.57% ]
lldap  | 2024-02-19T18:56:32.279737489Z 2024-02-19T18:56:32.189223596+00:00  DEBUG    │  │  ┝━ 🐛 [debug]:  | request.base: "ou=people,dc=example,dc=nz" | scope: Users
lldap  | 2024-02-19T18:56:32.279744623Z 2024-02-19T18:56:32.189226524+00:00  DEBUG    │  │  ┝━ get_user_list [ 935µs | 0.02% / 0.44% ]
lldap  | 2024-02-19T18:56:32.279750738Z 2024-02-19T18:56:32.189257385+00:00  DEBUG    │  │  │  ┝━ 🐛 [debug]:  | filters: And([And([And([]), And([And([]), MemberOf("ocis_users")]), Not(UserId(UserId("false")))]), UserId(UserId("adam"))])
lldap  | 2024-02-19T18:56:32.279757960Z 2024-02-19T18:56:32.189265069+00:00  DEBUG    │  │  │  ┕━ list_users [ 885µs | 0.42% ] filters: Some(And([And([And([]), And([And([]), MemberOf("ocis_users")]), Not(UserId(UserId("false")))]), UserId(UserId("adam"))])) | _get_groups: false
lldap  | 2024-02-19T18:56:32.279768669Z 2024-02-19T18:56:32.190937607+00:00  DEBUG    │  │  │     ┕━ 🐛 [debug]:  | return: [UserAndGroups { user: User { user_id: UserId("adam"), email: "adam@example.nz", display_name: Some("Adam Shand"), creation_date: 2023-05-19T09:37:38.282695250, uuid: Uuid("c5c8724a-bfaa-36d0-ae77-6cc936bf5808"), attributes: [AttributeValue { name: "avatar", value: Serialized("hash: 0xEDB02F50A5FB7F1D") }, AttributeValue { name: "first_name", value: Serialized("Adam") }, AttributeValue { name: "last_name", value: Serialized("Shand") }] }, groups: Some([GroupDetails { group_id: GroupId(5), display_name: "calibre", creation_date: 2023-05-20T05:15:14.956410978, uuid: Uuid("59463dc6-5700-3d0d-96bc-432a5aa43f5f"), attributes: [] }, GroupDetails { group_id: GroupId(4), display_name: "jellyfin-admin", creation_date: 2023-05-20T03:35:06.456309899, uuid: Uuid("e5c0de8a-25a7-37f1-be06-eb82b97e7d6d"), attributes: [] }, GroupDetails { group_id: GroupId(1), display_name: "lldap_admin", creation_date: 2023-05-19T09:35:39.078795764, uuid: Uuid("100fcf69-388b-3725-889f-8c496f9cec40"), attributes: [] }, GroupDetails { group_id: GroupId(6), display_name: "ocis_users", creation_date: 2024-02-19T01:08:57.207719922, uuid: Uuid("4afbf888-08c1-336a-8c79-45a7c4570060"), attributes: [] }]) }]
lldap  | 2024-02-19T18:56:32.279785740Z 2024-02-19T18:56:32.191568379+00:00  DEBUG    │  │  ┝━ expand_attribute_wildcards [ 13.3µs | 0.01% ] ldap_attributes: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"]
lldap  | 2024-02-19T18:56:32.279792693Z 2024-02-19T18:56:32.191581211+00:00  DEBUG    │  │  │  ┕━ 🐛 [debug]:  | resolved_attributes: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"]
lldap  | 2024-02-19T18:56:32.279798975Z 2024-02-19T18:56:32.191595559+00:00  WARN     │  │  ┕━ 🚧 [warn]: Ignoring unrecognized group attribute: text\n\
lldap  | 2024-02-19T18:56:32.279805325Z                       To disable this warning, add it to "ignored_user_attributes" in the config.
lldap  | 2024-02-19T18:56:32.279688737Z 2024-02-19T18:56:32.189192027+00:00  INFO     ┝━ LDAP request [ 1.43ms | 0.10% / 0.67% ]
lldap  | 2024-02-19T18:56:32.279694940Z 2024-02-19T18:56:32.189211273+00:00  DEBUG    │  ┝━ 🐛 [debug]:  | msg: LdapMsg { msgid: 2, op: SearchRequest(LdapSearchRequest { base: "ou=people,dc=example,dc=nz", scope: Subtree, aliases: Never, sizelimit: 1, timelimit: 60, typesonly: false, filter: And([And([Equality("objectclass", "inetOrgPerson"), And([Equality("objectclass", "person"), Equality("memberOf", "cn=ocis_users,ou=groups,dc=example,dc=nz")]), Not(Equality("uid", "FALSE"))]), Equality("uid", "adam")]), attrs: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"] }), ctrl: [] }
lldap  | 2024-02-19T18:56:32.279729418Z 2024-02-19T18:56:32.189213744+00:00  DEBUG    │  ┝━ do_search [ 1.22ms | 0.13% / 0.57% ]
lldap  | 2024-02-19T18:56:32.279737489Z 2024-02-19T18:56:32.189223596+00:00  DEBUG    │  │  ┝━ 🐛 [debug]:  | request.base: "ou=people,dc=example,dc=nz" | scope: Users
lldap  | 2024-02-19T18:56:32.279744623Z 2024-02-19T18:56:32.189226524+00:00  DEBUG    │  │  ┝━ get_user_list [ 935µs | 0.02% / 0.44% ]
lldap  | 2024-02-19T18:56:32.279750738Z 2024-02-19T18:56:32.189257385+00:00  DEBUG    │  │  │  ┝━ 🐛 [debug]:  | filters: And([And([And([]), And([And([]), MemberOf("ocis_users")]), Not(UserId(UserId("false")))]), UserId(UserId("adam"))])
lldap  | 2024-02-19T18:56:32.279757960Z 2024-02-19T18:56:32.189265069+00:00  DEBUG    │  │  │  ┕━ list_users [ 885µs | 0.42% ] filters: Some(And([And([And([]), And([And([]), MemberOf("ocis_users")]), Not(UserId(UserId("false")))]), UserId(UserId("adam"))])) | _get_groups: false
lldap  | 2024-02-19T18:56:32.279768669Z 2024-02-19T18:56:32.190937607+00:00  DEBUG    │  │  │     ┕━ 🐛 [debug]:  | return: [UserAndGroups { user: User { user_id: UserId("adam"), email: "adam@example.nz", display_name: Some("Adam Shand"), creation_date: 2023-05-19T09:37:38.282695250, uuid: Uuid("c5c8724a-bfaa-36d0-ae77-6cc936bf5808"), attributes: [AttributeValue { name: "avatar", value: Serialized("hash: 0xEDB02F50A5FB7F1D") }, AttributeValue { name: "first_name", value: Serialized("Adam") }, AttributeValue { name: "last_name", value: Serialized("Shand") }] }, groups: Some([GroupDetails { group_id: GroupId(5), display_name: "calibre", creation_date: 2023-05-20T05:15:14.956410978, uuid: Uuid("59463dc6-5700-3d0d-96bc-432a5aa43f5f"), attributes: [] }, GroupDetails { group_id: GroupId(4), display_name: "jellyfin-admin", creation_date: 2023-05-20T03:35:06.456309899, uuid: Uuid("e5c0de8a-25a7-37f1-be06-eb82b97e7d6d"), attributes: [] }, GroupDetails { group_id: GroupId(1), display_name: "lldap_admin", creation_date: 2023-05-19T09:35:39.078795764, uuid: Uuid("100fcf69-388b-3725-889f-8c496f9cec40"), attributes: [] }, GroupDetails { group_id: GroupId(6), display_name: "ocis_users", creation_date: 2024-02-19T01:08:57.207719922, uuid: Uuid("4afbf888-08c1-336a-8c79-45a7c4570060"), attributes: [] }]) }]
lldap  | 2024-02-19T18:56:32.279785740Z 2024-02-19T18:56:32.191568379+00:00  DEBUG    │  │  ┝━ expand_attribute_wildcards [ 13.3µs | 0.01% ] ldap_attributes: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"]
lldap  | 2024-02-19T18:56:32.279792693Z 2024-02-19T18:56:32.191581211+00:00  DEBUG    │  │  │  ┕━ 🐛 [debug]:  | resolved_attributes: ["dn", "uid", "mail", "displayName", "sn", "givenName", "entryuuid", "text"]
lldap  | 2024-02-19T18:56:32.279798975Z 2024-02-19T18:56:32.191595559+00:00  WARN     │  │  ┕━ 🚧 [warn]: Ignoring unrecognized group attribute: text\n\
lldap  | 2024-02-19T18:56:32.279805325Z                       To disable this warning, add it to "ignored_user_attributes" in the config.
```

Reply:
@FirstYear, I think we got ourselves a parsing bug in ldap3_proto: it seems they're trying to use an attribute option (like in http://www.ldapexplorer.com/en/manual/107070400-binary-attributes.htm) but we're parsing that as an extra attribute. I'm not sure where exactly in the rfc it's described, but the attribute "uuid;text" should resolve to just "uuid" with the text option.

For this PR, no action item, we'll merge the config as is.